src/main/scala/chiseltest/package.scala

```diff
@@ -422,8 +423,14 @@ package object chiseltest {
       inner
     }
 
-    def enumToString(record: EnumType): BigInt => String = {
-      def inner(bits: BigInt): String = "[unimplemented enum decode]"
+    def enumToString(tpe: EnumType): BigInt => String = {
+      def inner(bits: BigInt): String = {
+        val fullName = chisel3.internaltest.EnumHelpers.valueToName(tpe, bits).getOrElse("???")
+        // we only want to class and value name, not the package or enclosing class
+        val noPackage = fullName.split('.').takeRight(2).mkString(".")
```

Comment:
Is the full name usually package.chiselModuleName.enumObjectName.enumValue ?
Are there cases where there may be nested (dynamically created) classes inside a chisel module (or any nested set of classes) like package.chiselModuleName.factoryClass.enumObjectName.enumValue in which case we want the enclosing class?
Since this is just a debugging feature - it's not a big deal to get this precisely right in every case

Reply:
There might be exceptions. I would just wait until someone complains.